File: LeetCodeProblemsCMake/Problems/LeetCode2918MinimumEqualSumOfTwoArraysAfterReplacingZeros.cpp
//
// Created by saeed on 5/10/25.
//

#include "LeetCode2918MinimumEqualSumOfTwoArraysAfterReplacingZeros.h"
using namespace std;

long long LeetCode2918MinimumEqualSumOfTwoArraysAfterReplacingZeros::minSum(vector<int> &nums1, vector<int> &nums2) {
    int firstZeroCount{};
    long long firstSum{};
    const int firstLength = static_cast<int>(nums1.size());
    int secondZeroCount{};
    long long secondSum{};
    const int secondLength = static_cast<int>(nums2.size());
    for (int i = 0; i < firstLength; i++) {
        if (nums1[i] == 0) {
            firstZeroCount++;
            continue;
        }
        firstSum += nums1[i];
    }
    for (int i = 0; i < secondLength; i++) {
        if (nums2[i] == 0) {
            secondZeroCount++;
        }
        secondSum += nums2[i];
    }
    firstSum+=firstZeroCount;
    secondSum+=secondZeroCount;
    if (firstZeroCount==0) {
        if (firstSum<secondSum) {
            return -1;
        }
    }
    if (secondZeroCount==0) {
        if (secondSum<firstSum) {
            return -1;
        }
    }
    return max(firstSum, secondSum);
}

File: LeetCodeProblemsGo/LeetCode2918MinimumEqualSumOfTwoArraysAfterReplacingZeros/LeetCode2918MinimumEqualSumOfTwoArraysAfterReplacingZeros.go
package main

func minSum(nums1 []int, nums2 []int) int64 {
	firstZeroCount := 0
	var firstSum int64 = 0
	firstLength := len(nums1)
	secondZeroCount := 0
	var secondSum int64 = 0
	secondLength := len(nums2)
	for i := 0; i < firstLength; i++ {
		if nums1[i] == 0 {
			firstZeroCount++
			continue
		}
		firstSum += int64(nums1[i])
	}
	for i := 0; i < secondLength; i++ {
		if nums2[i] == 0 {
			secondZeroCount++
			continue
		}
		secondSum += int64(nums2[i])
	}
	firstSum += int64(firstZeroCount)
	secondSum += int64(secondZeroCount)
	if firstZeroCount == 0 {
		if firstSum < secondSum {
			return -1
		}
	}
	if secondZeroCount == 0 {
		if secondSum < firstSum {
			return -1
		}
	}
	if firstSum < secondSum {
		return secondSum
	}
	return firstSum
}
